mlflow/utils/logging_utils.py

```diff
@@ -87,3 +89,31 @@ def _configure_mlflow_loggers(root_module_name):
 
 def eprint(*args, **kwargs):
     print(*args, file=MLFLOW_LOGGING_STREAM, **kwargs)  # pylint: disable=print-function
+
+
+class LoggerMessageFilter(logging.Filter):
+    def __init__(self, module: str, filter_regex: str):
+        super().__init__()
+        self._pattern = re.compile(filter_regex)
+        self._module = module
+
+    def filter(self, record):
+        if record.name == self._module and self._pattern.search(record.msg):
+            return False
+        return True
+
+
+@contextlib.contextmanager
+def suppress_logs(module: str, filter_regex: str):
```

Comment:
```suggestion
def suppress_logs(module: str, filter_regex: re.Pattern):
```

can we pre-compile the regex?